feat: add a depends-on conditions (#2619)
Mergifyio/mergify-engine: docs/source/actions/merge.rst

         itself as the commit message. The pull request number will be added to
         end of the title.

.. _queue-depends-on:

⛓️ Defining Pull Request Dependencies
-------------------------------------

|premium plan tag|
|open source plan tag|

You can specify dependencies between pull requests from the same repository.
Mergify waits for the linked pull requests to be merged before merging any pull

> request with a ``Depends-on:`` header.

Reviewer: ```suggestion
request with a ``Depends-on:`` header.
```
```suggestion
request with a ``Depends-On:`` header.
```

To use this feature, adds the ``Depends-On:`` header to the body of your pull
request:

.. code-block:: md

    New awesome feature 🎉

    To get the full picture, you may need to look at these pull requests:

    Depends-On: #42
    Depends-On: https://github.com/organization/repository/pull/123

> This feature does not work for cross repository dependencies.

Reviewer: ```suggestion
.. warning::

    This feature does not work for cross-repository dependencies.
```

         itself as the commit message. The pull request number will be added to
         end of the title.

.. _queue-depends-on:

⛓️ Defining Pull Request Dependencies
-------------------------------------

|premium plan tag|
|open source plan tag|

You can specify dependencies between pull requests from the same repository.
Mergify waits for the linked pull requests to be merged before merging any pull
request with a ``Depends-On:`` header.

To use this feature, adds the ``Depends-On:`` header to the body of your pull
request:

.. code-block:: md

    New awesome feature 🎉

    To get the full picture, you may need to look at these pull requests:

    Depends-On: #42
    Depends-On: https://github.com/organization/repository/pull/123

.. warning::

>     This feature does not work for cross repository dependencies.

Reviewer: ```suggestion
    This feature does not work for cross-repository dependencies.
```

         itself as the commit message. The pull request number will be added to
         end of the title.

.. _depends-on:

> ⛓️ Defining the pull request dependencies

Reviewer: ```suggestion
⛓️ Defining Pull Request Dependencies
```

----------------------------------------

> |premium plan tag|

Reviewer: ```suggestion
|open source plan tag|
|premium plan tag|
```

> Mergify can wait for other pull requests to be merged before merging this one.
> 
> To do this you must add a ``Depends-On:`` lines to you pull request body:

Reviewer: ```suggestion
You can specify dependencies between pull requests from the same repository. Mergify waits for the linked pull requests to be merged before merging any pull request with a ``Depends-on:`` header.

To use this feature, adds the ``Depends-On:`` header to the body of your pull request:
```
